.github/workflows/android-build.yml

```diff
@@ -0,0 +1,52 @@
+name: Build Sample for Android
+
+on:
+  push
+
+jobs:
+  apk:
+    name: Apk Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+              sudo apt update
+              sudo apt install openjdk-8-jdk ant android-sdk-platform-tools-common android-sdk ninja-build
+      - name: Setup Path
+        run: |
+              PATH="/usr/src/android-sdk/platform-tools:$PATH"
+              export ANDROID_HOME="/usr/src/android-sdk"
+      - name: Install and setup SDKManager
+        run: |
+              wget https://dl.google.com/android/repository/commandlinetools-linux-6609375_latest.zip
+              unzip commandlinetools-linux-6609375_latest.zip -d cmdline-tools
+              mkdir --parents "$ANDROID_HOME/cmdline-tools/latest"
+              sudo mv cmdline-tools/* "$ANDROID_HOME/cmdline-tools/latest/"
+              export PATH=$ANDROID_HOME/cmdline-tools/latest/bin:$PATH
+              sdkmanager --sdk_root=$ANDROID_HOME --install "platforms;android-31" "build-tools;31.0.0"
+              yes | sdkmanager --sdk_root=$ANDROID_HOME --licenses
+      - name: Clone the project and prepare the build
+        run: |
+              git clone -b android-build-with-cmake https://github.com/captain1947/sdl3-sample --depth=1 --recurse-submodules

```

Comment:
I think I addressed this one, though I am not sure if I did what you wanted. I just used ref to change the branch instead of using -b flag with git clone. Also, we have to remove the branch stuff before merging as your repo only have a single main branch. But if we do that now, the workflow will fail in my repo and I won't be able to test.